jgiven-core/src/main/java/com/tngtech/jgiven/impl/inject/ValueInjector.java

```diff
@@ -42,57 +43,65 @@ public void validateFields( Object object ) {
 
         Map<Object, Field> resolvedFields = Maps.newHashMap();
 
-        for( Field field : getScenarioFields( object ) ) {
-            field.setAccessible( true );
-            Resolution resolution = getResolution( field );
+        for( ScenarioStateField field : getScenarioFields( object ) ) {
```

Comment:
Now we go through annotations each time `getScenarioFields` is called, which still happens multiple times even within one method. I think refactoring this would deserve its own PR, though, in particular since this has been existng behavior.